src/main/java/org/haffson/adventofcode/days/day06/Coordinate.java

```diff
@@ -0,0 +1,358 @@
+package org.haffson.adventofcode.days.day06;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Helper record for day 6.
+ * Is used to work better with the string inputs.
+ * Coordinates have an x and y value.
+ * Also, you can compare two coordinates with each other.
+ */
+public record Coordinate(int xCoordinate, int yCoordinate) {
+
+    @Nonnull
+    public static Coordinate of(@Nonnull final String input) {
+        final String[] split1 = input.split(",");
+        final String[] split2 = split1[1].split(" ");
+
+        final int xCoordinate = Integer.parseInt(split1[0]);
+        final int yCoordinate = Integer.parseInt(split2[1]);
+
+        return new Coordinate(xCoordinate, yCoordinate);
+    }
+
+    /**
+     * Helper method for day 6.
+     * Gives a list<Coordinate> of a string-list.
+     *
+     * @return List<Coordinate>
+     */
+    @Nonnull
+    public static List<Coordinate> of(@Nonnull final List<String> stringList) {
+        return stringList.stream().map(Coordinate::of).toList();
+    }
+
+    /**
+     * Helper method for day 6.
+     * Compares the distance of two coordinates.
+     *
+     * @return int of how many fields are between two coordinates.
+     */
+    public static int coordinateFieldDistance(@Nonnull final Coordinate coordinateToCompare,
+                                              @Nonnull final Coordinate comparedCoordinate) {
+        int xDistance = Math.abs(coordinateToCompare.xCoordinate - comparedCoordinate.xCoordinate);
+        int yDistance = Math.abs(coordinateToCompare.yCoordinate - comparedCoordinate.yCoordinate);
```

Comment:
Du solltest, wann immer möglich, `final` nutzen.

```suggestion
        final int xDistance = Math.abs(coordinateToCompare.xCoordinate - comparedCoordinate.xCoordinate);
        final int yDistance = Math.abs(coordinateToCompare.yCoordinate - comparedCoordinate.yCoordinate);
```

Reply:
Das merke ich jetzt nicht weiter an, aber es gilt für den gesamten Code.

Reply:
Der Haken bei IntelliJ war tatsächlich draußen ?! Werde nochmal überall drüber schauen